portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Policies/Policies.tsx

```diff
@@ -175,53 +177,83 @@ const Policies: React.FC = () => {
         const commonPoliciesPromise = API.getCommonOperationPolicies();
         Promise.all([apiPoliciesPromise, commonPoliciesPromise]).then((response) => {
             const [apiPoliciesResponse, commonPoliciesResponse] = response;
-            const apiSpecificPolicies = apiPoliciesResponse.body.list;
-            const commonPolicies = commonPoliciesResponse.body.list;
-            const mergedList = [...commonPolicies, ...apiSpecificPolicies];
+            const apiSpecificPoliciesList = apiPoliciesResponse.body.list;
+            const commonPoliciesList = commonPoliciesResponse.body.list;
+            const mergedList = [...commonPoliciesList, ...apiSpecificPoliciesList];
 
             // Get all common policies and API specific policies
             setAllPolicies(mergedList);
 
-            let unionByPolicyDisplayName;
+            let apiPolicyByPolicyDisplayName;
+            let commonPolicyByPolicyDisplayName;
             if (showMultiVersionPolicies) {
```

Comment:
This can be simplified as follows.

```javascript
// Get the policies depending on the policy display name and version
apiPolicyByPolicyDisplayName = [...apiSpecificPoliciesList
    .reduce((map: Map<string, Policy>, obj: Policy) => 
        map.set((showMultiVersionPolicies ? obj.name + obj.version : obj.name), obj), new Map()).values()];
commonPolicyByPolicyDisplayName = [...commonPoliciesList
    .reduce((map: Map<string, Policy>, obj: Policy) => 
        map.set((showMultiVersionPolicies ? obj.name + obj.version : obj.name), obj), new Map()).values()];
```